File: code/python/714.py
class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        dp_i_0 = 0
        dp_i_1 = float('-inf')
        for price in prices:
            temp = dp_i_0
            dp_i_0 = max(dp_i_0,dp_i_1+price)
            dp_i_1 = max(dp_i_1,temp-price-fee)
        return dp_i_0

File: code/java/714.java
class Solution {
    public int maxProfit(int[] prices, int fee) {
        int dp_i_0 = 0;
        int dp_i_1 = -Integer.MAX_VALUE;
        int temp = -1;
        for(int price:prices){
            temp = dp_i_0;
            dp_i_0 = Math.max(dp_i_0,dp_i_1+price);
            dp_i_1 = Math.max(dp_i_1,temp-price-fee);
        }
        return dp_i_0;
    }
}